model/user.go

```diff
@@ -48,45 +48,48 @@ const (
+	Id                         string    `json:"id"`
+	CreateAt                   int64     `json:"create_at,omitempty"`
+	UpdateAt                   int64     `json:"update_at,omitempty"`
+	DeleteAt                   int64     `json:"delete_at"`
+	Username                   string    `json:"username"`
+	Password                   string    `json:"password,omitempty"`
+	AuthData                   *string   `json:"auth_data,omitempty"`
+	AuthService                string    `json:"auth_service"`
+	Email                      string    `json:"email"`
+	EmailVerified              bool      `json:"email_verified,omitempty"`
+	Nickname                   string    `json:"nickname"`
+	FirstName                  string    `json:"first_name"`
+	LastName                   string    `json:"last_name"`
+	Position                   string    `json:"position"`
+	Roles                      string    `json:"roles"`
+	AllowMarketing             bool      `json:"allow_marketing,omitempty"`
+	Props                      StringMap `json:"props,omitempty"`
+	NotifyProps                StringMap `json:"notify_props,omitempty"`
+	LastPasswordUpdate         int64     `json:"last_password_update,omitempty"`
+	LastPictureUpdate          int64     `json:"last_picture_update,omitempty"`
+	FailedAttempts             int       `json:"failed_attempts,omitempty"`
+	Locale                     string    `json:"locale"`
+	Timezone                   StringMap `json:"timezone"`
+	MfaActive                  bool      `json:"mfa_active,omitempty"`
+	MfaSecret                  string    `json:"mfa_secret,omitempty"`
+	LastActivityAt             int64     `db:"-" json:"last_activity_at,omitempty"`
+	AcceptedServiceTermsId     string    `json:"accepted_service_terms_id,omitempty"`
+	LatestServiceTermsAccepted *bool     `db:"-" json:"latest_service_terms_accepted,omitempty"`
 }
 
 type UserPatch struct {
-	Username    *string   `json:"username"`
-	Nickname    *string   `json:"nickname"`
-	FirstName   *string   `json:"first_name"`
-	LastName    *string   `json:"last_name"`
-	Position    *string   `json:"position"`
-	Email       *string   `json:"email"`
-	Props       StringMap `json:"props,omitempty"`
-	NotifyProps StringMap `json:"notify_props,omitempty"`
-	Locale      *string   `json:"locale"`
-	Timezone    StringMap `json:"timezone"`
+	Username               *string   `json:"username"`
+	Nickname               *string   `json:"nickname"`
+	FirstName              *string   `json:"first_name"`
+	LastName               *string   `json:"last_name"`
+	Position               *string   `json:"position"`
+	Email                  *string   `json:"email"`
+	Props                  StringMap `json:"props,omitempty"`
+	NotifyProps            StringMap `json:"notify_props,omitempty"`
+	Locale                 *string   `json:"locale"`
+	Timezone               StringMap `json:"timezone"`
+	AcceptedServiceTermsId string    `json:"accepted_service_terms_id"`
```

Comment:
Do we actually ever need to patch this field? If not, I suggest leaving it out of the `UserPatch` struct altogether.

Reply:
no we don't. Removed.